pyroute2/netlink/rtnl/rtmsg.py

```diff
@@ -160,7 +160,144 @@ class seg6_encap_info(nla):
         __slots__ = ()
 
         nla_map = (('SEG6_IPTUNNEL_UNSPEC', 'none'),
-                   ('SEG6_IPTUNNEL_SRH', 'hex'))
+                   ('SEG6_IPTUNNEL_SRH', 'ipv6_sr_hdr'))
+
+        class ipv6_sr_hdr(nla):
+
+            __slots__ = ()
+
+            fields = (('encapmode', 'I'),
+                      ('nexthdr', 'B'),
+                      ('hdrlen', 'B'),
+                      ('type', 'B'),
+                      ('segments_left', 'B'),
+                      ('first_segment', 'B'),
+                      ('flags', 'B'),
+                      ('reserved', 'H'),
+                      ('segs', 's'),
+                      # Potentially several type-length-value
+                      ('tlvs', 's'))
```

Comment:
I am not sure if this is the better way to do that. What do you think ?

Reply:
Probably there is no better way right now, you're welcome to propose. This one also may not always work as expected, but I'll fix that.